determined-ai/determined
# fix: localize debounced settings updates

## Ticket
ET-287

## Description
This pr solves some race conditions that were added in #9220 by undoing the global settings debounce and instead localizing it to the column width changes the bug was talking about. There's a light refacator of `userSettings#setPartial` which should allow it to work on any io-ts type that has props.



## Test Plan
1. Open your browsers devtools and navigate to the network tab.
2. navigate the page to the new experiment list.
3. change the sort order either by using the column header menu or the dropdown menu. Note that there should be one POST request to settings per change.
4. change the width of any column. Only one POST request should be made when the column width is done moving.



## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.




[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
